lib/internal/util.js

```diff
@@ -203,6 +203,42 @@ function getConstructorOf(obj) {
   return null;
 }
 
+function getIdentifiableTagOf(obj) {
+  const original = obj;
+  let constructor = '';
+  let tag = '';
+
+  while (obj) {
+    if (constructor === '') {
+      var desc = Object.getOwnPropertyDescriptor(obj, 'constructor');
+      if (desc !== undefined &&
+         typeof desc.value === 'function' &&
+         desc.value.name !== '')
+        constructor = desc.value.name;
+    }
+
+    if (tag === '') {
+      desc = Object.getOwnPropertyDescriptor(obj, Symbol.toStringTag);
+      if (desc !== undefined) {
+        if (desc.value !== undefined)
+          tag = desc.value;
+        else if (desc.get !== undefined)
+          tag = desc.get.call(original);
+      }
+    }
+
+    if (constructor !== '' && tag !== '')
+      break;
+
+    obj = Object.getPrototypeOf(obj);
+  }
+
+  if (constructor === tag)
+    return constructor;
+
+  return `${constructor} ${tag ? `[${tag}]` : ''}`.trim();
```

Comment:
I think we could instead just branch on `tag` to avoid the `trim()`. Something like:

```js
if (tag)
  return `${constructor} [${tag}]`;

return constructor;
```